lib/src/main/java/io/ably/lib/rest/Auth.java

```diff
@@ -617,10 +617,15 @@ else if(!request.keyName.equals(keyName))
 
 		/* timestamp */
 		if(request.timestamp == 0) {
-			if(options.queryTime)
-				request.timestamp = ably.time();
-			else
+			if (options.queryTime) {
+				/* Spec: RSA10k */
+				if (timeOffset == null) {
+					timeOffset = ably.time() - timestamp();
+				}
+				request.timestamp = timeOffset + timestamp();
+			} else {
 				request.timestamp = timestamp();
```

Comment:
So the flow should look like this:
1) customer initiates a single request with queryTime = 1 to get synced with server time;
2) all subsequent requests should happen with queryTime = 0 but use timeOffset calculated in the initial request?
3) if new request with queryTime = 1 initiated we need to get synced with server time once again?

Right behavior?

Current implementation was done with assumption that all requests which should rely on server time and offset should have queryTime = 1 and use discardTimeOffset when required to query time once again. Look like it was not a correct assumption. Please confirm.